lightly/active_learning/utils/object_detection_output.py

```diff
@@ -0,0 +1,98 @@
+""" TODO """
+
+# Copyright (c) 2020. Lightly AG and its affiliates.
+# All Rights Reserved
+
+from typing import List
+
+from lightly.active_learning.utils.bounding_box import BoundingBox
+
+
+class ObjectDetectionOutput:
+    """Class which unifies different object detection output formats.
+
+    Attributes:
+        boxes:
+            List of BoundingBox objects with coordinates (x0, y0, x1, y1).
+        scores:
+            List of confidence scores (i.e. max(class prob) * objectness).
+        labels:
+            List of labels.
+
+    Examples:
+        >>> # typical model output
+        >>> prediction = {
+        >>>     'boxes': [[0.1, 0.2, 0.3, 0.4]],
+        >>>     'scores': [0.1234],
+        >>>     'labels': [1]
+        >>> }
+        >>>
+        >>> # convert bbox to objects
+        >>> boxes = [BoundingBox(0.1, 0.2, 0.3, 0.4)]
+        >>> scores = prediction['scores']
+        >>> labels = prediction['labels']
+        >>>
+        >>> # create detection output
+        >>> detection_output = ObjectDetectionOutput(boxes, scores, labels)
+
+    """
+
+    def __init__(self,
+                 boxes: List[BoundingBox],
+                 scores: List[float],
+                 labels: List[int]):
```

Comment:
Why use labels instead of the class probabilities? I thought we discussed that we use class probs, bounding boxes and object probs

Reply:
Yes, the intuition here was that you can always go from class probabilities to labels/scores but you can't go the other way. If we require class probabilities there are many people who can't use our scorer. Additionally, we don't have any scorers yet which require the class probability so I thought we could add it as an option later?

Reply:
I feel we set here a standard. Switching it later might be complicated. We can always simplify (e.g. replace class probs with argmax internally). But the other way around might be breaking, no?